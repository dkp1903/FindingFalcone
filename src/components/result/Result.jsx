import React, {Component} from 'react';
import PropTypes from 'prop-types';
import './Result.scss';

const Result = (props) => {
    console.log('Prop status: ', props.status)
    if(props.status == "success") 
      return (
        
          <div className="result">
            <h1>Found Falcone!</h1>
            <p>You got me the damn Falcone!</p>
            <p>Time Taken: {props.count}</p>
            <p>Planet Found: {props.planetName}</p>
            <button onClick={() => window.location.reload()}>Not satisfied? Try your luck again</button>
          </div>  
        
      ); 
    else 
      return (
      
         <div className="result">
          <h1>Not found Falcone!</h1>
          
          <p>Time Taken: {props.count}</p>
          
          <button onClick={() => window.location.reload()}>Retry</button>
        </div>
    )
  
}

Result.propTypes = {
  count: PropTypes.number.isRequired,
  planetName: PropTypes.string.isRequired,
}

export default Result