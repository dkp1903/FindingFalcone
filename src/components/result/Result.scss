.result{
  text-align: center;
  padding: 20px;

  h1{
    margin-bottom: 80px;
  }

  p{
    font-size: 18px;
    font-weight: 600;
    margin-bottom: 25px;
  }

  button{
    background-color: #5850ec;
    border-radius: 10px;
    color: white;
    font-family: 'Poppins';
    padding: 15px 32px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 16px;
  }
}
