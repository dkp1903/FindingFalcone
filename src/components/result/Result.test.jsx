import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import Result from "./Result.jsx";

let container = null;

beforeEach(() => {
  
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("check render", () => {
  act(() => {
    render(<Result count={200} planetName="PlanetB" />, container);
  });
  expect(container).toMatchSnapshot();
});
