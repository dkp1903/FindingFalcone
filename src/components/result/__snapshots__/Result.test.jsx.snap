// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders successfully with props 1`] = `
<div>
  <div
    class="result"
  >
    <h1>
      Not found Falcone!
    </h1>
    <p>
      Time Taken: 
      200
    </p>
    <button>
      Retry
    </button>
  </div>
</div>
`;
